src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/HdcService.kt

```diff
@@ -1,23 +1,44 @@
 package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service
 
+import jakarta.persistence.EntityNotFoundException
 import jakarta.transaction.Transactional
 import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.HdcCurfewTimesRepository
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.LicenceRepository
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.hdc.HdcApiClient
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.hdc.HdcLicenceData
 
 @Service
 class HdcService(
   private val hdcApiClient: HdcApiClient,
+  private val hdcCurfewTimesRepository: HdcCurfewTimesRepository,
+  private val licenceRepository: LicenceRepository,
 ) {
 
   @Transactional
-  fun getHdcLicenceData(bookingId: Long): HdcLicenceData? {
-    val licenceData = this.hdcApiClient.getByBookingId(bookingId)
+  fun getHdcLicenceData(licenceId: Long): HdcLicenceData? {
+    val licence = licenceRepository
+      .findById(licenceId)
+      .orElseThrow { EntityNotFoundException("No licence data found for $licenceId") }
 
-    return HdcLicenceData(
-      curfewAddress = licenceData.curfewAddress,
-      firstNightCurfewHours = licenceData.firstNightCurfewHours,
-      curfewHours = licenceData.curfewHours,
-    )
+    val curfewTimes = hdcCurfewTimesRepository.findByLicenceId(licenceId)
+
+    val licenceData = this.hdcApiClient.getByBookingId(licence.bookingId!!)
```

Comment:
I think we still need to call the `hdcApiClient` every time as we don't yet store `curfewAddress` or `firstNightCurfewHours` in the CVL db